src/components/entity/ha-entity-picker.ts

```diff
@@ -168,18 +172,14 @@ export class HaEntityPicker extends LitElement {
           this.includeEntities!.includes(entityId)
         );
 
-        return entityIds
-          .map((key) => ({
+        return entityIds.map((key) => {
+          const friendly_name = computeStateName(hass!.states[key]) || key;
+          return {
             ...hass!.states[key],
-            friendly_name: computeStateName(hass!.states[key]) || key,
-          }))
-          .sort((entityA, entityB) =>
```

Comment:
You removed the sorting by name?

Reply:
`fuzzyFilterSort` should be responsible for sorting. However, I guess it could also make sense to keep the extra name sort for the case before `fuzzyFilterSort` has been called.